src/data/repositories/NotificationD2Repository.ts

```diff
@@ -6,37 +6,117 @@ import {
+                notifiedAlert =>
+                    notifiedAlert.outbreak === outbreakName &&
+                    notifiedAlert.district === alert.district
+            );
+            if (notifiedAlertExists) {
+                return Future.success(undefined);
+            }
+
+            const updatedNotifiedAlerts = [
+                ...notifiedAlerts,
+                { district: alert.district, outbreak: outbreakName },
+            ];
+
+            return Future.joinObj({
+                districtName: this.getDistrictName(alert.district),
+                saveNotification: this.saveNotifiedAlertsToDataStore(updatedNotifiedAlerts),
+            }).flatMap(({ districtName }) => {
+                return apiToFuture(
+                    this.api.messageConversations.post({
+                        subject: `New Outbreak Alert: ${outbreakName} in ${districtName}`,
+                        text: buildNotificationText(
+                            outbreakName,
+                            districtName,
+                            notificationOptions
+                        ),
+                        userGroups: userGroups,
+                    })
+                ).flatMap(() => Future.success(undefined));
+            });
+        });
+    }
+
+    private getDistrictName(districtId: string): FutureData<string> {
         return apiToFuture(
-            this.api.messageConversations.post({
-                subject: `New Outbreak Alert: ${outbreakName} in zm Zambia Ministry of Health`,
-                text: buildNotificationText(outbreakName, notificationOptions),
-                userGroups: userGroups,
+            this.api.metadata.get({
+                organisationUnits: {
+                    fields: {
+                        name: true,
+                    },
+                    filter: {
+                        id: {
+                            eq: districtId,
+                        },
+                    },
+                },
             })
-        ).flatMap(() => Future.success(undefined));
+        )
+            .flatMap(response => assertOrError(response.organisationUnits[0], "Organisation Unit"))
+            .map(district => district.name);
+    }
+
+    private getNotifiedAlertsFromDataStore(): FutureData<NotifiedAlert[]> {
+        return this.dataStoreClient
+            .getObject<NotifiedAlert[]>(notifiedAlertsDatastoreKey)
+            .map(notifiedAlerts => notifiedAlerts ?? []);
+    }
+
+    private saveNotifiedAlertsToDataStore(notifiedAlerts: NotifiedAlert[]): FutureData<void> {
+        return this.dataStoreClient.saveObject<NotifiedAlert[]>(
+            notifiedAlertsDatastoreKey,
+            notifiedAlerts
+        );
     }
 }
 
-function buildNotificationText(outbreakKey: string, notificationData: NotificationOptions): string {
-    const { detectionDate, emergenceDate, incidentManager, notificationDate, verificationStatus } =
-        notificationData;
+function buildNotificationText(
```

Comment:
Plase, change in this function the params `outbreakKey` --> `outbreakName` and `district` --> `districtName` for a better understanding